src/main/java/com/backbase/ct/bbfuel/configurator/BillPayConfigurator.java

```diff
@@ -32,4 +49,67 @@ public void ingestBillPayUser(String externalUserId) {
             log.info("Bill Pay user [{}] ingested", externalUserId);
         }
     }
+
+    public void ingestBillPayAccounts(String subscriberId, List<Account> accounts) {
+        loginRestClient.login(subscriberId, subscriberId);
+        userContextPresentationRestClient.selectContextBasedOnMasterServiceAgreement();
+        UserByIdPutRequestBody request = new UserByIdPutRequestBody()
+            .withSubscriber(createSubscriber(subscriberId, accounts));
+        Response response = billPayMockBankRestClient.ingestAccounts(request);
+        if (isBadRequestException(response, "The request is invalid")) {
+            log.info("Bad request for Bill Pay user [{}]", subscriberId);
+        } else {
+            response.then().assertThat().statusCode(SC_OK);
+            log.info("Accounts ingested for Bill Pay user [{}]", subscriberId);
+        }
+    }
+
+    public void ingestBillPayUserAndAccounts(LegalEntityWithUsers le, Boolean ingestAccounts) {
+        List<Account> accounts = new ArrayList<>();
+        if (ingestAccounts) {
+            String externalUserId = le.getUserExternalIds().get(0);
+            loginRestClient.login(externalUserId, externalUserId);
+            String externalLegalEntityId = this.userPresentationRestClient.
+                retrieveLegalEntityByExternalUserId(externalUserId)
+                .getExternalId();
+            List<ArrangementItem> arrangementItems = accountsIntegrationRestClient
+                .getArrangements(externalLegalEntityId);
+
+            arrangementItems.forEach((arrangement) -> {
+                accounts.add(mapBillPayAccount(arrangement));
+            });
+        }
+        le.getUserExternalIds().forEach((user) -> {
+            ingestBillPayUser(user);
+            if (ingestAccounts) {
+                ingestBillPayAccounts(user, accounts);
+            }
+        });
+    }
+
+    private Subscriber createSubscriber(String subscriberId, List<Account> accounts) {
+        return new Subscriber()
+            .withUser(subscriberId)
+            .withSubscriberId(subscriberId)
+            .withFirstName("John")
+            .withLastName("Doe")
+            .withTaxIdentifier("tax00001")
+            .withAddress(
+                new Address()
+                    .withAddress1("560 Carillon Parkway")
+                    .withAddress2("Carillon")
+                    .withCity("Saint Petersburg")
+                    .withState("FL")
+                    .withPostalCode("33717")
+            )
+            .withAccounts(accounts);
+    }
+
+    public Account mapBillPayAccount(ArrangementItem arrangement) {
+        Random random = new Random();
+        return new Account()
+            .withAccountNumber(arrangement.getBBAN())
+            .withRoutingNumber(arrangement.getBankBranchCode())
+            .withAccountType(AccountType.values()[random.nextInt(AccountType.values().length)]);
```

Comment:
you could rewrite this with some help of `CommonHelper` : `withAccountType(getRandomFromList(asList(AccountType.values())))`